bootloader/src/sd_update_progress.c

```diff
@@ -0,0 +1,59 @@
+#include "sd_update_progress.h"
+
+#include <furi_hal.h>
+
+static const char* update_state_text[] = {
```

Comment:
Despite of the fact that we use this construction in some other places, I prefer having separatemethod that returns string by enum and have switch inside of it.


In the worst case use something like this:

```

static const Icon* ArchiveItemIcons[] = {
    [ArchiveFileTypeIButton] = &I_ibutt_10px,
    [ArchiveFileTypeNFC] = &I_Nfc_10px,
    [ArchiveFileTypeSubGhz] = &I_sub1_10px,
    [ArchiveFileTypeLFRFID] = &I_125_10px,
    [ArchiveFileTypeIrda] = &I_ir_10px,
    [ArchiveFileTypeFolder] = &I_dir_10px,
    [ArchiveFileTypeUnknown] = &I_unknown_10px,
};
```

Reply:
In my pet project, I have found that the following kind of construction is most suitable for such cases.

```
static const char* error_texts[] = {
    [FR_OK] = "OK",
    [FR_DISK_ERR] = "DISK ERR",
    [FR_INT_ERR] = "INT ERR",
    [FR_NOT_READY] = "NOT READY",
    [FR_NO_FILE] = "NO FILE",
    [FR_NO_PATH] = "NO PATH",
    [FR_INVALID_NAME] = "INVALID NAME",
    [FR_DENIED] = "DENIED",
    [FR_EXIST] = "EXIST",
    [FR_INVALID_OBJECT] = "INVALID OBJECT",
    [FR_WRITE_PROTECTED] = "WRITE PROTECTED",
    [FR_INVALID_DRIVE] = "INVALID DRIVE",
    [FR_NOT_ENABLED] = "NOT ENABLED",
    [FR_NO_FILESYSTEM] = "NO FILESYSTEM",
    [FR_MKFS_ABORTED] = "MKFS ABORTED",
    [FR_TIMEOUT] = "TIMEOUT",
    [FR_LOCKED] = "LOCKED",
    [FR_NOT_ENOUGH_CORE] = "NOT ENOUGH CORE",
    [FR_TOO_MANY_OPEN_FILES] = "TOO MANY OPEN FILES",
    [FR_INVALID_PARAMETER] = "INVALID PARAMETER",
};

const char* fs_error_text(uint32_t error) {
    if(error >= COUNT_OF(error_texts)) {
        return "UNKNOWN";
    } else {
        return error_texts[error];
    }
}
```